### PR TITLE
B1

### PR DESCRIPTION
**Pull request by kethzigilbert**
I have changed the naming of few components, cut down some code and implemented conditional rendering in the same component rather than having a seperate component

### CODE REVIEW

**Comment by kethzigilbert** [comment]
Updated StarRating Component and removed redux design.

File: src/Components/DisplayBooks.js [hunk 1]
```diff
@@ -1,45 +1,61 @@
-import React from 'react';
-import {ListGroupItem,ListGroup, Card, CardBody, CardHeader, CardFooter, CardTitle, CardSubtitle} from 'reactstrap';
-import StarRating from './RenderStarRating';
+import React, { lazy, Suspense } from 'react';
+import { Card, CardBody, CardFooter, CardTitle, CardSubtitle } from 'reactstrap';
+import StarRating from './StarRating';
+const DisplayBooksSecondPart = lazy(() => import('./DisplayBooksSecondPart'), 'default')
+
+function bookDisplayCards(book) {
```
Comment (AmeyKamat) [inline src/Components/DisplayBooks.js]: Is it possible to extract this a separate component and just use it as per requirement?

Rule of thumb: Any jsx redering logic should not be outside render method

File: src/Components/DisplayBooks.js [hunk 2]
```diff
@@ -1,45 +1,61 @@
...
+
+
+
+
+    );
+}
+
+const DisplayBooks = (props) => {
+
+    var books1to12 = props.books.slice(0, 12);
```
Comment (AmeyKamat) [inline src/Components/DisplayBooks.js]: Can we name variables more suitably?
Also avoid var at all costs

File: src/Components/DisplayBooks.js [hunk 3]
```diff
@@ -1,45 +1,61 @@
...
+    var books2 = booksRest.map(bookDisplayCards);
+
+    return (
+        <React.Fragment>
+            
+            
+                {books1}
+           
+
+            <Suspense fallback={<h3>Still Loading…</h3>}>
```
Comment (AmeyKamat) [inline src/Components/DisplayBooks.js]: What will happen if number of books < 12

File: src/Components/DisplayBooks.js [hunk 4]
```diff
@@ -1,45 +1,61 @@
...
+
+    );
+}
+
+const DisplayBooks = (props) => {
+
+    var books1to12 = props.books.slice(0, 12);
+    var booksRest = props.books.slice(12, props.books.length);
+    var books1 = books1to12.map(bookDisplayCards);
+    var books2 = booksRest.map(bookDisplayCards);
```
Comment (AmeyKamat) [inline src/Components/DisplayBooks.js]: This whole logic can be performed inside return statement by chaining function calls

File: src/Components/DisplayBooksSecondPart.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React, { Component} from 'react';
+class DisplayBooksSecondPart extends Component{
```
Comment (AmeyKamat) [inline src/Components/DisplayBooksSecondPart.js]: Name this component more suitably

File: src/Components/MainComponent.js [hunk 6]
```diff
@@ -1,35 +1,30 @@
...
-      <Header/>
-      
-      <Book/>
-
-            
-             
-      <br/>
-      </div>
-      <Footer/>
+        <div className="taskbody_background">
```
Comment (AmeyKamat) [inline src/Components/MainComponent.js]: Camelcase

File: src/Components/StarRating.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+const StarRating = (props) => {
+  
+let starElements =[];
+let rating = Math.floor(props.rating);
+    
+for ( var i=1 ; i<=5; i++){
```
Comment (AmeyKamat) [inline src/Components/StarRating.js]: Use map function instead inside return statement

File: src/redux/ActionCreators.js [hunk 8]
```diff
@@ -1,18 +1,16 @@
...
-
 export const BooksLoading = () => ({
     type: ActionTypes.BOOKS_LOADING
 });
-
-
 export const fetchBooks = () => (dispatch) => {
-    dispatch(BooksLoading(true));
+    dispatch(BooksLoading());
     return fetch(baseUrl ,
```
Comment (AmeyKamat) [inline src/redux/ActionCreators.js]: It is good practice to hold all api calls separately in its own js file and make it reusable.

File: src/Components/MainComponent.js (hunk 6 above)
Comment (kethzigilbert) [reply src/Components/MainComponent.js]: This is a css class .

File: src/redux/ActionCreators.js (hunk 8 above)
Comment (kethzigilbert) [reply src/redux/ActionCreators.js]: not using this file in the application.

**Pull request merged**